### ISSUE TITLE
Where is the "y_pred" calculated?

### ISSUE THREAD

**Issue by yyaodong** [fchollet/keras#236]
Hi Fchollet:

I intend to change the loss function in the model.compile, and I find that in the file of objective functions, the "y_pred" is regarded as input. I want to know where is this "y_pred" calculated? For example, in softmax regression,  it is the process of y_pred = T.argmax(self.p_y_given_x, axis=1).

Cheers,
YYD

**Comment by fchollet** [comment]
`y_pred` is the output of the last layer of your network (i.e. it is the prediction of the network). Its dimensionality is determined by that last layer.

**Comment by yyaodong** [comment]
Thanks for your reply.

Let's say I use the softmax layer in the last. The output of that activation layer is T.nnet.softmax(x) which is not the "y_pred" yet. I am wondering in which process does the last layer compute the T.argmax(self.p_y_given_x, axis=1)?

Thanks

**Comment by harrismirza** [comment]
I am also facing the same issue, the output of my network is a dense layer with 590 units a Softmax activation, but y_pred just returns a list of 590 numbers with the largest as 1 and the rest as 0, what is the reason for this? Should it not just return the prediction from the network?
